atlp-rwanda/dynamites-ecomm-fe
# Implement Role-Based Access Control

This PR adds role-based access control (RBAC) to the frontend of the project, allowing for fine-grained access management based on user roles.

### **Tasks Completed:**

- [x] Defined roles and permissions for vendors, buyers, and admins.
- [x] Integrated an API endpoint for managing roles and permissions.
- [x] Enforced role-based access control throughout the frontend application.
- [x] Implemented appropriate error handling and user feedback for access control issues.
- [x] Added comprehensive unit tests for the RBAC implementation.
### **How to Test:**

- Clone the repository and set up the development environment.
- Perform CRUD operations on roles by interacting with the /api/v1/roles route via the frontend interface.
- Test the checkRole and checkPermissions middleware by accessing routes that require specific roles and permissions.
- Verify that users with different roles see the appropriate UI elements and have access to the correct features.
### **Context:**
- There are three roles: Vendor, Buyer, and Admin.

### **Vendor Permissions:**

- Manage Products
- Fulfill Orders
- View Sales Reports
- Manage Inventory
- Manage Account
### **Buyer Permissions:**

- Browse Products
- Add to Cart
- Check Out
-  Order History
- Track Orders
- Manage Account
- Leave Reviews
- View Wishlist
- Receive Notifications
### **Admin Permissions:**

- Manage Users
- Manage Roles
- Manage Products
- Manage Orders
- Manage Categories
- Manage Shipping
- Manage Payments
